orttraining/orttraining/test/training_ops/cpu/nn/dropout_op_test.cc

```diff
@@ -109,6 +109,101 @@ void RunDropoutTest(const char* op, const bool use_mask, const std::vector<int64
 
   t.Run(OpTester::ExpectResult::kExpectSuccess, "", {}, nullptr, nullptr, ExecutionMode::ORT_SEQUENTIAL, output_verifier);
 }
+
+void RunBiasDropoutTest(const char* op, const bool use_mask, const std::vector<int64_t>& input_shape, float ratio = -1,
+                        bool training_mode = true, bool use_float16_ratio = false, bool has_residual = true) {
+  OpTester t{op, 1, kMSDomain};
+  const int64_t seed = 42;
+  t.AddAttribute("seed", seed);
+
+  const auto input_size = std::accumulate(
+      input_shape.begin(), input_shape.end(), static_cast<int64_t>(1), std::multiplies<>{});
+  std::vector<float> input(input_size);
+  std::iota(input.begin(), input.end(), 1.0f);
+  t.AddInput("data", input_shape, input);
+
+  std::vector<int64_t> bias_shape{input_shape.back()};
+  const auto bias_size = input_shape.back();
+  std::vector<float> bias(bias_size);
+  std::iota(bias.begin(), bias.end(), 2.0f);
+  t.AddInput("bias", bias_shape, bias);
+
+  float residual_value = 0.0f;
+  const auto residual_size = input_size;
+  std::vector<float> residual(residual_size);
+  if (has_residual) {
+    residual_value = 1.0f;
+    std::fill(residual.begin(), residual.end(), residual_value);
+    t.AddInput("residual", input_shape, residual);
+  } else {
+    t.AddMissingOptionalInput<float>(); 
+  }
+
+  if (ratio == -1) {
+    ratio = 0.5;  // default.
+    t.AddInput("ratio", {}, {ratio});
+  } else if (use_float16_ratio) {
+    t.AddInput("ratio", {}, {MLFloat16(0)});
+  } else {
+    t.AddInput("ratio", {}, {ratio});
+  }
+
+  if (training_mode) {
+    t.AddInput("training_mode", {}, {true});
+  }
+
+  t.AddOutput<float>("output", input_shape, input);  // we'll do our own output verification
+
+  std::unique_ptr<bool[]> mask_buffer{};
+  if (use_mask) {
+    mask_buffer = onnxruntime::make_unique<bool[]>(input_size);
+    t.AddOutput<bool>("mask", input_shape, mask_buffer.get(), input_size);
+  } else {
+    t.AddMissingOptionalOutput<bool>();
+  }
+
+  auto output_verifier = [&](const std::vector<OrtValue>& fetches, const std::string& provider_type) {
+    ASSERT_GE(fetches.size(), 1);
+    const auto& output_tensor = FetchTensor(fetches[0]);
+    auto output_span = output_tensor.DataAsSpan<float>();
+
+    const auto num_output_zeros = std::count(output_span.begin(), output_span.end(), residual_value);
```

Comment:
>num_output_zeros [](start = 15, length = 16)

seems like this could either be number of ones or number of zeroes depending on has_residual